File: S2M1/src/FactoriaCarreraCarretera.java
public class FactoriaCarreraCarretera implements FactoriaCarreraYBicicleta {

	public Carrera crearCarrera() {
		return new CarreraCarretera();
	}

	public Bicicleta crearBicicleta(int dorsal, boolean tengoAcabar) {
		return new BicicletaCarretera(dorsal, tengoAcabar);
	}
}

File: S2M1/Ruby/FactoriaCarreraCarretera.rb
class FactoriaCarreraCarretera  < FactoriaCarreraYBicicleta

	def crearCarrera()
		return CarreraCarretera.new()
    end

	def crearBicicleta(dorsal, tengoAcabar) 
		return BicicletaCarretera.new(dorsal, tengoAcabar)
    end
end